src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -140,36 +271,54 @@ public RunVersionOne CreateRun(Run v2Run)
 
                 if (v2Run.TryGetProperty("sarifv1/run", out serializedV1Run))
                 {
-                    run = JsonConvert.DeserializeObject<RunVersionOne>(serializedV1Run, Utilities.JsonSettingsV1);
+                    run = JsonConvert.DeserializeObject<RunVersionOne>(Regex.Unescape(serializedV1Run), Utilities.JsonSettingsV1Compact);
```

Comment:
Bug: The call to`Regex.Unescape` isn't right.

You put the run into the property bag by calling `JsonConvert.SerializeObject`, which escapes the set of characters that need to be escaped in a JSON string. Then you take it out of the property bag by calling `Regex.Unescape`, which unescapes the set of characters that need to be escaped in a regex.

But those are two different sets of characters.

Reply:
Since it's double-encoded, I don't know how to get back to a valid json string. Using Regex.Unescape was the approach recommended by others who were doing the same thing.

Reply:
Could you explain what you mean by "double-encoded"?

Reply:
"Escaped" is what I really mean. Without the Unescape call, the quotes are still escaped with backslashes in the string obtained by TryGetProperty..

Reply:
We need to think more about this. By my reasoning above (different sets of escaped characters), this approach is just wrong, and you're unlucky that it works most of the time (unlucky, because it won't fail for months or years, and then we'll struggle to figure out why).

Code that is _almost_ correct is more dangerous than code that's obviously wrong and fails immediately ;-)

Reply:
Yes, for this reason I spent a good deal of time trying to find the right approach. I'm all ears :-)

Reply:
Changed to this:
    RunVersionOne run = null;
    if (v2Run.TryGetProperty<RunVersionOne>("sarifv1/run", out run))
    {
        return run;
    }

Results in this exception:
Newtonsoft.Json.JsonSerializationException : Error converting value "{"tool":{"name":"CodeScanner","semanticVersion":"2.1.0"},"results":[]}" to type 'Microsoft.CodeAnalysis.Sarif.VersionOne.RunVersionOne'. Path '', line 1, position 84.
---- System.ArgumentException : Could not cast or convert from System.String to Microsoft.CodeAnalysis.Sarif.VersionOne.RunVersionOne.

Reply:
I wouldn't expect that to work, because your property value is a string, not a `run` object.

Reply:
I guess I could store the object itself?
      "properties": {
        "sarifv2/run": {"tool":{"name":"AssetScanner","semanticVersion":"1.7.2"},"results":[]}
      }

Reply:
You mean, as opposed to storing a serialized version of the object? That might work. Why did we originally decide to store the serialized string? It was probably to preserve whitespace on round-tripping. But if we're not doing that anyway, then this might work.

Reply:
It just seemed natural to store it serialized.
This seems to be working. The only thing I'm seeing is enums are emitted as their integer value rather than their string name.

Reply:
But they deserialize correctly

Reply:
Here's my stupid suggestion, base64 encode the string. Does it really matter to provide a 'readable' embedded JSON file in the property bag? Probably not. This definitely sidesteps all the horrible escaping issues implied by what we're attempting here.

Reply:
@EasyRhin0 @michaelcfanning This exploration looks very promising! I'll have to sign off for the evening in a few minutes, but I'll look in tomorrow and see where we landed. Good work!

Reply:
(BTW, you are absolutely right that it doesn't matter how enum values are stored in the property bag, as long as they round-trip correctly.)

Reply:
And I'm intrigued by Michael's base64 idea. It does indeed side-step all the escaping/unescaping nonsense -- but at a cost of doubling (is that right?) the size of the bagged property. Really I like Chris's "store the object, not the serialized string" approach best. Its only downsize is not preserving whitespace, which I've given up on :-)